File: app/src/androidTest/java/com/example/rafaelanastacioalves/moby/util/HelperMethods.java
package com.example.rafaelanastacioalves.moby.util;

import android.support.test.espresso.matcher.BoundedMatcher;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import org.hamcrest.Description;
import org.hamcrest.Matcher;

public class HelperMethods {

    public static Matcher<RecyclerView.ViewHolder> withHolderContainingId(final int id) {
        return new BoundedMatcher<RecyclerView.ViewHolder, RecyclerView.ViewHolder>(RecyclerView.ViewHolder.class) {

            @Override
            protected boolean matchesSafely(RecyclerView.ViewHolder item) {
                View visualizeView = item.itemView.findViewById(id);
                if (visualizeView == null) {
                    return false;
                }
                return true;
            }

            @Override
            public void describeTo(Description description) {
                description.appendText("No ViewHolder found with id: " + id);
            }

        };
    }
}

File: app/src/androidTest/java/com/example/rafaelanastacioalves/moby/util/HelperMethods.kt
package com.example.rafaelanastacioalves.moby.util

import android.support.test.espresso.matcher.BoundedMatcher
import android.support.v7.widget.RecyclerView
import android.view.View

import org.hamcrest.Description
import org.hamcrest.Matcher

object HelperMethods {

    fun withHolderContainingId(id: Int): Matcher<RecyclerView.ViewHolder> {
        return object : BoundedMatcher<RecyclerView.ViewHolder, RecyclerView.ViewHolder>(RecyclerView.ViewHolder::class.java) {

            override fun matchesSafely(item: RecyclerView.ViewHolder): Boolean {
                val visualizeView = item.itemView.findViewById<View>(id) ?: return false
                return true
            }

            override fun describeTo(description: Description) {
                description.appendText("No ViewHolder found with id: $id")
            }

        }
    }
}
